File: CodeForces/Div.4/#886/D/main.cpp
#include <bits/stdc++.h>

using namespace std;

#define endl '\n'
typedef long long i64;
typedef int i32;
typedef short i16;
typedef vector<i64> vi64;
typedef vector<vi64> vv;
typedef string str;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  i64 t;
  cin >> t;

  while (t--) {
    i64 n, k;
    cin >> n >> k;

    vi64 v(n);
    for (i64 i = 0; i < n; i++) {
      cin >> v[i];
    }

    sort(v.begin(), v.end());
    i64 crt = 1, res = -1;

    for (i64 i = 1; i < n; i++) {
      if (v[i] - v[i - 1] <= k) {
        crt++;
      } else {
        if (crt > res) {
          res = crt;
        }
        crt = 1;
      }
    }

    if (crt > res) {
      res = crt;
    }

    if (res == -1) {
      cout << 0 << endl;
    } else {
      cout << n - res << endl;
    }
  }

  return 0;
}

File: CodeForces/Div.4/#886/D/main.py
def main():
    t = int(input())

    for i in  range(t):
        n, k = input().split()
        n = int(n)
        k = int(k)
        l = list(map(int, input().split()))
        l.sort()

        crt, res = 1, -1
        for ii in range(1, n):
            if l[ii] - l[ii - 1] <= k:
                crt += 1
            else:
                if crt > res:
                    res = crt
                crt = 1

        if crt > res:
            res = crt

        if res == -1:
            print(0)
        else:
            print(n - res)

main()
             